docs/source/download.rst

```diff
@@ -67,11 +57,41 @@ venv_ as above and then run::
   where ``LOCALHOSTNAME`` is the name returned by running the `hostname`
   command. Should the local host name change, this may require updating.
 
+Debugging install problems
+--------------------------
+
+If ``firedrake-install`` fails, the following flowchart describes some
+common build problems and how to solve them. If you understand the
+prognosis and feel comfortable making these fixes yourself then great!
+If not, feel free to ask for more help in our
+:doc:`Slack channel </contact>`.
+
+.. graphviz:: install-debug.dot
+
+If you don't see the issue you're experiencing in this chart, please
+ask us on Slack or report a bug by creating a new `github issue
+<https://github.com/firedrakeproject/firedrake/issues>`__. To help us
+diagnose what's going wrong, **please include the following log files**:
+
+* ``firedrake-install.log`` from Firedrake
+* ``configure.log`` and ``make.log`` from PETSc
+
+Likewise, if it's ``firedrake-update`` that fails, please include the
+file ``firedrake-update.log``. The PETsc log files live in the
+directory ``src/petsc/`` inside the directory where the Firedrake
+virtual environment was created.
```

Comment:
Can you also make it clear exactly where `firedrake-install.log` and `firedrake-update.log` can be found?